WeekFormulas.ts

```diff
@@ -203,27 +207,64 @@ function protectWeek(weekNum: number): void {
         wplProtection = getColumn(weekNum, 8).protect().setDescription("WPL PROTECTION");
     const sheetEditors = wahfProtection.getEditors();
 
-    wahfProtection.removeEditors(sheetEditors).addEditors(databaseAdmin);
-    semProtection
-        .removeEditors(sheetEditors)
-        .addEditors(
-            databaseAdmin.concat(seminarFiller.tls.map((tl) => tl.email))
-        );
-    ssProtection
-        .removeEditors(sheetEditors)
-        .addEditors(
-            databaseAdmin.concat(studySessionFiller.tls.map((tl) => tl.email))
-        );
-    fdProtection
-        .removeEditors(sheetEditors)
-        .addEditors(
-            databaseAdmin.concat(frontDeskFiller.tls.map((tl) => tl.email))
-        );
-    zoomProtection.removeEditors(sheetEditors).addEditors(databaseAdmin);
-    tlmProtection.removeEditors(sheetEditors).addEditors(databaseAdmin);
-    gamProtection
-        .removeEditors(sheetEditors)
-        .addEditors(databaseAdmin.concat(gaInfo));
-    mcfProtection.removeEditors(sheetEditors).addEditors(databaseAdmin);
-    wplProtection.removeEditors(sheetEditors).addEditors(databaseAdmin);
+    //grab row length
+    
+    const proctectedSheet = SpreadsheetApp.getActiveSpreadsheet().getSheetByName("Protection");
+
+    var range = getFormulaResult("=COUNTA(Protection!B1:1)");
```

Comment:
I think you can remove column A in the `Protection` sheet because it's redundant for what we have here. Then that should shift the columns. Also, you shouldn't start the count from the name of the protection, i.e. it shouldn't use `["Front Desk", "email1@gmail.com", ...]` for the count, but instead just `["email1@gmail.com",...]`.